decompiler/structures/ast/condition_symbol.py

```diff
@@ -80,7 +80,13 @@ def get_z3_condition_map(self) -> Dict[LogicCondition, PseudoLogicCondition]:
         return dict((symbol, condition_symbol.z3_condition) for symbol, condition_symbol in self._condition_map.items())
 
     def add_condition(self, condition: Condition) -> ConditionSymbol:
-        """Adds a condition to the condition map."""
+        """Adds a new condition to the condition map and returns the corresponding condition_symbol"""
+        for value in self._condition_map.values():
```

Comment:
We could also think about comparing the pseudo conditions, i.e., check whether they are equivalent. But not sure about runtime.

Reply:
It would reduce the number of symbols and would make comparisons easier. 
But the only method which compares symbols is `removeGuardedDoWhile`....

So yeah maybe not that worth for now.